core/commands/cat.go

```diff
@@ -50,7 +51,20 @@ var CatCmd = &cmds.Command{
 			return
 		}
 
-		readers, length, err := cat(req.Context(), node, req.Arguments(), int64(offset))
+		max, found, err := req.Option("length").Int()
```

Comment:
USE int64. THIS COULD OVERFLOW

Reply:
I don't see a int64 option in go-ipfs-cmdkit, we could use a uint instead? We could also add a new option if not.

Reply:
I was really just giving @Stebalien crap, he was sitting behind me while i reviewed this and pointed out that this should be an int64